### DISCUSSION TITLE
Calculation of diametral inertia

### DISCUSSION THREAD

**Discussion by lilian-tec** [petrobras/ross#1025] [General]
Hello, 

I have a definition problem with the diametral moment of inertia.
How is it calculated?

`disk1 = rs.DiskElement.from_geometry(
    n=4,
    material=steel,
    width=0.07,
    i_d=0.05,
    o_d=0.28
)`

> Mass           (kg):      32.59
Diam. inertia  (kg*m**2): 0.17809
Polar. inertia (kg*m**2): 0.32956

Re-creating the geometry in a CAD model, you get

> Ix = 0.2180116989456501 # kg.m2
Iy = 0.2180116989456501 # kg.m2
Iz = 0.3295636208913719 # kg.m2

I suppose you get Id from Ix and Iy but cannot found out how, my guess was that Id = Ix = Iy

Thank you!

**Comment by lilian-tec** [comment]
To follow up on that:

The reason is that I am doing a side-by-side comparison ANSYS v ROSS.
It is a pump shaft with an overhang impeller.

The results match 1-on-1 when the shaft only is modelled.
Once I add the impeller, the results start to diverge, as in ROSS shows a large divergence between the forward and backward mode, and so far I have not been able to reproduce it.

Any ideas is appreciated!

![image](https://github.com/petrobras/ross/assets/117370500/e9b75a92-3c1f-4e53-aefe-37cdf74e36b9)

**Comment by raphaeltimbo** [comment]
Hi @lilian-tec!
The polar moment of inertia about the shaft ($I_p$) and the diametral moment of inertia about any axis perpendicular to the shaft line ($I_d$) are calculated in this part of the code:

https://github.com/petrobras/ross/blob/be780255d7a63a4b88786fb5a88deff10044926c/ross/disk_element.py#L430-L437

I will double check this formula to see if there are any issues.

**Comment by raphaeltimbo** [comment]
Hi, @lilian-tec.
I have just opened the #1029 PR to improve our documentation regarding the properties calculation for the disk. We now have the following in the documentation:

![image](https://github.com/petrobras/ross/assets/18506378/dbaeae0b-0e91-485f-9625-b5aafd360600)

I have also altered the formulas in the code to match the reference/documentation just to make this verification process easier (they will give the same results as before).

Therefore, I think this part of the code is working as expected and we would need to investigate the differences in the Campbell diagram elsewhere.
